examples/most-redis-docker/config.js

```diff
@@ -0,0 +1,14 @@
+// Functions to handle app configuration.
+
+const config
+  = exports.config
+  = env => {
+    const port = Number(env.REDIS_PORT || 6379)
+    const host = String(env.REDIS_HOST || '127.0.0.1')
+    const timeout = Number(env.TIMEOUT || 2500)
+    const channel = env.REDIS_CHANNEL
+
+    if (!channel) throw new Error(`REDIS_CHANNEL not specified.`)
```

Comment:
Since this is an example, how do you feel about making the channel default do something that's likely unique, like `most-redis-docker-example`, rather than failing?

Reply:
I'm on the fence.  On the one hand, it's very frustrating to get cryptic errors when trying to run an example.  

On the other hand, [early detection of configuration errors is important](https://blog.acolyer.org/2016/11/29/early-detection-of-configuration-errors-to-reduce-failure-damage/) for microservices.  I think it's also important for example apps:

- ✅ Defaulting to the wrong port or host will fail loudly and fast.  
- ✅ Defaulting to the wrong timeout will not affect 99.99% of the users.
- ❌ Defaulting to the wrong channel will fail silently, _causing user frustration_.

I'll create a `validate` function, inject a logging function, and emit a wordy, helpful message if the user skips the REDIS_CHANNEL.  I'll also provide a default so it doesn't crash.  I did not like how this `config` function wasn't separating concerns, anyways. 😄 

Sound good?

Reply:
Ok, so the validate function got a bit unwieldy for an example app.  I just injected a logging function and used that instead of throwing.

Reply:
Good points.  I was thinking that for an example, it'd be ok to default.  I like the idea of adding logging in that case.  Ultimately, I'm cool with either:

* defaulting and logging which channel is being used (whether default or not), and highlighting when it has defaulted because of REDIS_CHANNEL not being specified.
* keeping the `throw` and updating the error message to say what the user needs to do to fix the issue.  Something like: "REDIS_CHANNEL not specified.  Please set the REDIS_CHANNEL env var to the name of a valid redis channel that this example should use, such as 'most-redis-docker-example'"